lib/core/Axios.js

```diff
@@ -58,6 +60,11 @@ Axios.prototype.request = function request(config) {
 
 Axios.prototype.getUri = function getUri(config) {
   config = mergeConfig(this.defaults, config);
+
+  if (config.baseURL && !isAbsoluteURL(config.url)) {
+    config.url = combineURLs(config.baseURL, config.url);
```

Comment:
This mutates the configuration object,  which will cause issues with subsequent requests. Any mutations to `url`  should be done in a local variable. e.g.,

```
let url = config.url;
if ( .... ) {
   url = combineURLs( ... );
}

return buildURL(url, ...);
```

Reply:
‘mergeConfig’ returns a new configuration object, which means the config passed to ‘getUri’ is not mutated when we set the url.

Reply:
Yikes. `mergeConfig` itself sounds like a big piece of tech debt that needs to go. It should be possible to structure this library such that we dont have to copy a fat object like that so many times. It's frequently a source of bugs, as experienced with `retry-axios` ( https://github.com/JustinBeckwith/retry-axios/issues/1  https://github.com/JustinBeckwith/retry-axios/issues/50  https://github.com/JustinBeckwith/retry-axios/issues/4  ) cc @emilyemorehouse Curious to hear your thoughts around `mergeConfig`